src/ast_utils.h

```diff
@@ -46,6 +46,34 @@ struct BreakSeeker : public PostWalker<BreakSeeker, Visitor<BreakSeeker>> {
   }
 };
 
+// Finds all functions that are reachable via direct calls.
+
+struct DirectCallGraphAnalyzer : public PostWalker<DirectCallGraphAnalyzer, Visitor<DirectCallGraphAnalyzer>> {
+  Module *module;
+  std::vector<Function*> queue;
+  std::unordered_set<Function*> reachable;
+
+  DirectCallGraphAnalyzer(Module* module, std::vector<Function*> root) : module(module) {
+    for (auto curr : root) {
```

Comment:
Not necessary, but it does add a compiler check that the type here is in fact a pointer. Imagine if root is changed to be a list of some object that is heavy to copy, then this line would error on such a change if it has `auto*`, but silently be slow without.